### PR TITLE
[DX] Check php version features parameter type

### PR DESCRIPTION
**Pull request by TomasVotruba**
Closes https://github.com/rectorphp/rector/issues/6734

## Before  :bug: 

The `rector.php` configuration:

```php
$parameters->set(Option::PHP_VERSION_FEATURES, '10000');

// or
$parameters->set(Option::PHP_VERSION_FEATURES, 7.4);
```

...silent pass...

<br>

## Now :rocket: 

Screenshot from 2021-10-09 21-08-11

<br><br>

```php
$parameters->set(Option::PHP_VERSION_FEATURES, \Rector\Core\ValueObject\PhpVersion::PHP_54);
```

:+1:

### CODE REVIEW

File: src/Php/PhpVersionProvider.php [hunk 1]
```diff
@@ -48,4 +52,27 @@ public function isAtLeastPhpVersion(int $phpVersion): bool
...
         return $phpVersion <= $this->provide();
     }
+
+    private function validatePhpVersionFeaturesParameter(mixed $phpVersionFeatures): void
+    {
+        if ($phpVersionFeatures === null) {
+            return;
+        }
+
+        if (is_int($phpVersionFeatures)) {
```
Comment (staabm) [inline src/Php/PhpVersionProvider.php]: Maybe more precise using a regex like https://github.com/rectorphp/rector-src/pull/982/files#diff-17fe9c0401cf68f0174b184b240c505e82e480deee7a373217f71750f3e50108R91

Comment (staabm) [reply src/Php/PhpVersionProvider.php]: Or at least check for a int with at least 5 digits

Comment (TomasVotruba) [reply src/Php/PhpVersionProvider.php]: It makes sense.
Maybe we can go forward with enum value check :+1:

Comment (staabm) [reply src/Php/PhpVersionProvider.php]: The user can input any version he likes - as long as valid, no?

File: src/Php/PhpVersionProvider.php [hunk 2]
```diff
@@ -22,7 +24,9 @@ public function __construct(
 
     public function provide(): int
     {
-        $phpVersionFeatures = $this->parameterProvider->provideIntParameter(Option::PHP_VERSION_FEATURES);
+        $phpVersionFeatures = $this->parameterProvider->provideParameter(Option::PHP_VERSION_FEATURES);
```
Comment (samsonasik) [inline src/Php/PhpVersionProvider.php:34]: how about using `PhpVersionFactory->createIntVersion()` if the value is string? 

https://github.com/rectorphp/rector-src/blob/3c0cb22a58e6b8dc4195631046ba4a62ceabf279/src/Util/PhpVersionFactory.php#L9

that's the '7.2' will still working.

File: src/Php/PhpVersionProvider.php (hunk 1 above)
Comment (TomasVotruba) [reply src/Php/PhpVersionProvider.php]: Valid and meaninful. I think PHP 9999 does not make sense. We should also care about meaninful DX experience. That's what enums are for.

Comment (TomasVotruba) [reply src/Php/PhpVersionProvider.php]: But I get more what you mean from tests.

How about now? :)

File: src/Php/PhpVersionProvider.php (hunk 2 above)
Comment (TomasVotruba) [reply src/Php/PhpVersionProvider.php:34]: IMO changing types between strings and integers will cause many bugs, so we better avoid it.

Even better when we can use ENUM in a PHP config file with autocomplete :+1:

File: src/Php/PhpVersionProvider.php (hunk 1 above)
Comment (staabm) [reply src/Php/PhpVersionProvider.php]: thanks looks great <3

**Pull request merged**